core/src/main/java/org/mskcc/cbio/portal/util/GeneticProfileReader.java

```diff
@@ -84,7 +84,20 @@ public static GeneticProfile loadGeneticProfile(File file) throws IOException, D
              // For mutation data only we can have multiple files with the same genetic_profile.
              // There is a constraint in the mutation database table to prevent duplicated data
              // If this constraint is hit (mistakenly importing the same maf twice) MySqlBulkLoader will throw an exception
-             return existingGeneticProfile;
+
+             // make an object combining the pre-existing profile with the file-specific properties of the current file
+             GeneticProfile gp = new GeneticProfile(
+                     existingGeneticProfile.getStableId(),
+                     existingGeneticProfile.getCancerStudyId(),
+                     existingGeneticProfile.getGeneticAlterationType(),
+                     existingGeneticProfile.getDatatype(),
+                     existingGeneticProfile.getProfileName(),
+                     existingGeneticProfile.getProfileDescription(),
+                     existingGeneticProfile.showProfileInAnalysisTab());
+             gp.setGeneticProfileId(existingGeneticProfile.getGeneticProfileId());
+             gp.setTargetLine(geneticProfile.getTargetLine());
+             gp.setOtherMetadataFields(geneticProfile.getAllOtherMetadataFields());
+             return gp;
```

Comment:
Especially since this code is specifically for loading a second data file into the same mutation data profile, splitting the data over multiple files.